testrtc.py

```diff
@@ -1,27 +1,70 @@
-#!/usr/bin/python2.4
+#!/usr/bin/python
 #
 # Copyright 2014 Google Inc. All Rights Reserved.
 
-"""WebRTC Test
+"""WebRTC
 
-This module serves the WebRTC Test Page.
+This module serves the WebRTC troubleshooter page.
 """
-
 import cgi
 import logging
 import random
 import os
 import webapp2
+import urllib
+import datetime
+import re
+
+from google.appengine.ext import blobstore
+from google.appengine.ext.webapp import blobstore_handlers
 
 # Generate 10 kilobytes of random data and create a 10MB buffer from it.
 random_file = bytearray([random.randint(0,127) for i in xrange(0,10000)] * 1000)
 
+class UploadHandler(blobstore_handlers.BlobstoreUploadHandler):
+  # Generate and return blobstore upload URL.
+  def head(self):
+    referer = self.request.headers['Referer']
+    upload_url = blobstore.create_upload_url('/report/upload')
+    self.response.headers.add_header("Access-Control-Allow-Origin", referer)
+    self.response.headers['Response-Text'] = upload_url
+  # Upload file to the blobstore.
+  def post(self):
+    origin = self.request.headers['Origin']
+    fileName = self.request.headers['X-File-Name']
+    upload_files = self.get_uploads(field_name=fileName)
+    blob_info = upload_files[0]
+    self.response.headers.add_header("Access-Control-Allow-Origin", origin)
+    self.response.headers['Response-Text'] = origin + '/report/download/%s' % blob_info.key()
+
+class ServeHandler(blobstore_handlers.BlobstoreDownloadHandler):
+  def get(self, resource):
+    resource = str(urllib.unquote(resource))
+    blob_info = blobstore.BlobInfo.get(resource)
+    self.send_blob(blob_info)
+
 class TestDownloadFile(webapp2.RequestHandler):
   def get(self, size_kbytes):
     self.response.headers.add_header("Access-Control-Allow-Origin", "*")
     self.response.headers['Content-Type'] = 'application/octet-stream'
     self.response.out.write(random_file[0: int(size_kbytes)*1000])
-    
+
+class CleanBlobstore(blobstore_handlers.BlobstoreUploadHandler):
+  def get(self):
+    retentionPeriod = datetime.datetime.now() - datetime.timedelta(days=90)
+    gqlQuery = blobstore.BlobInfo.gql("WHERE creation <= :date",
+        date = datetime.datetime.now() <= retentionPeriod)
+    # Only guaranteed up to 1000 results at a time.
+    logFiles = gqlQuery.run()
+    for log in logFiles:
+      # Make sure to only delete log reports from the blobstore.
+      if re.match(r'testrtc-.*.log', log.filename):
+        blobstore.BlobInfo.get(log.key()).delete()
+        print log.filename
+
 app = webapp2.WSGIApplication([
+    ('/report/upload', UploadHandler),
+    ('/report/download/([^/]+)?', ServeHandler),
```

Comment:
Nit: naming. ServeHandler is so generic :/
Also I think report/id/action feels more natural for urls than report/action/id. (e.g. report/id would serve a nice rendered content allowing to inspect the bug... report/id/raw would serve the file ready to be downloaded).
```
('/report/new', NewReportHandler),
('/report/([^/]+)', ReportHandler),
```